src/sentry/api/endpoints/project_details.py

```diff
@@ -423,6 +425,28 @@ def validate_sensitiveFields(self, value):
     def validate_safeFields(self, value):
         return validate_pii_selectors(value)
 
+    def validate_targetSampleRate(self, value):
+        from sentry import features
+
+        organization = self.context["project"].organization
+        actor = self.context["request"].user
+        if not features.has("organizations:dynamic-sampling-custom", organization, actor=actor):
+            raise serializers.ValidationError(
+                "Organization does not have the custom dynamic sample rate feature enabled."
+            )
+
+        if organization.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT) != "project":
+            raise serializers.ValidationError(
+                "Must enable Manual Mode to configure project sample rates."
+            )
+
+        if not 0.0 <= value <= 1.0:
+            raise serializers.ValidationError(
+                "The targetSampleRate option must be between 0 and 1."
+            )
```

Comment:
```suggestion
        if not 0.0 <= value <= 1.0:
            raise serializers.ValidationError(
                "The targetSampleRate option must be between 0 and 1."
            )
```

Isn't this also checked by a serializer?